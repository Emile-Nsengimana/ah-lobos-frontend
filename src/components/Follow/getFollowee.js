import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { Link } from "react-router-dom";
import {
  followUser,
  unFollowUser,
  getFollowee
} from "../../redux/actions/followAction";
import UserProfile from "../profile/UserProfile";

export class getFollower extends Component {
  async componentDidMount() {
    const token = sessionStorage.getItem("token");
    await this.props.getFollowee(token);
  }

  render() {
    return (
      <>
        <div>
          <div className="getFollow">
            <div className="firstDiv" />
            <div>
              <div className="chooseFollow">
                <Link to="/followee">
                  <p className="followingParagraph">Following</p>
                </Link>
                <Link to="/follow">
                  <p>Followers</p>
                </Link>
                <hr color="#0D9BC6" id="lineFollowees" />
              </div>
              {this.props.followees !== undefined &&
                typeof this.props.followees !== "string" &&
                this.props.followees.map(follow => {
                  return (
                    <div className="followers" key={follow.username}>
                      <img
                        src={follow.image}
                        alt=""
                        className="FollowerImage"
                      />
                      <p id="nameFollow">
                        <span id="namesSize">
                          {follow.firstName} {follow.lastName}{" "}
                        </span>
                        <span className="emailShowing">{follow.email}</span>
                      </p>
                      <p className="followOrNot">
                        <span>@{follow.username} </span>
                        Following
                      </p>
                    </div>
                  );
                })}{" "}
              {typeof this.props.followees === "string" ? (
                <div className="zero-Followers">
                  You do not follow anyone yet.
                </div>
              ) : null}
            </div>
            <div className="ProfileDiv">
              <UserProfile profile={this.props.profile} />
            </div>
          </div>
        </div>
      </>
    );
  }
}

export const mapStateToProps = state => ({
  followees: state.getFollowee.followees,
  profile: state.profile
});
getFollower.propTypes = {
  getFollowee: propTypes.func,
  followees: propTypes.array,
  profile: propTypes.object
};

export default connect(
  mapStateToProps,
  { getFollowee, followUser, unFollowUser }
)(getFollower);
