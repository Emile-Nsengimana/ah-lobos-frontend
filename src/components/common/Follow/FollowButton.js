/* eslint-disable consistent-return */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { Link } from "react-router-dom";
import followImage from "../../../assets/images/follow.svg";
import unFollowImage from "../../../assets/images/unfollow.svg";
import {
  followUser,
  unFollowUser,
  getFollowee
} from "../../../redux/actions/followAction";

export class FollowButton extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClick = this.onClick.bind(this);
  }

  async onClick() {
    const token = sessionStorage.getItem("token");
    const msg = document.getElementById("warn-follow");
    if (!token) {
      msg.classList.add("show-it");
    }
    await this.props.followUser(token, this.props.author);
  }

  async unFollowonClick() {
    const token = sessionStorage.getItem("token");
    await this.props.unFollowUser(token, this.props.author);
  }

  render() {
    return (
      <>
        <button className="follow">
          {this.props.isFollowing ? (
            <img src={followImage} alt="" onClick={this.onClick} />
          ) : (
            <img src={unFollowImage} alt="" onClick={this.unFollowonClick} />
          )}
        </button>
        <div id="warn-follow" className="login-to-follow">
          Please{" "}
          <Link to={`/login?redirect=/article/${this.props.slug}`}> login</Link>
          to follow this user
        </div>
      </>
    );
  }
}
const mapStateToProps = state => ({
  follow: state.followAuser,
  unfollow: state.unFollowAuser,
  isFollowing: state.isFollowing
});

FollowButton.propTypes = {
  followUser: propTypes.func.isRequired,
  unFollowUser: propTypes.func.isRequired,
  isFollowing: propTypes.bool.isRequired,
  slug: propTypes.string,
  author: propTypes.string
};
export default connect(
  mapStateToProps,
  { followUser, unFollowUser, getFollowee }
)(FollowButton);
