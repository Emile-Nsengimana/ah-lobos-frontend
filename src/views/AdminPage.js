/* eslint-disable no-return-assign */
import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import jwt from "jsonwebtoken";
import Navbar from "../components/common/Navbar";
import Sidebar from "../components/common/Sidebar";
import Card from "../components/common/Card/Card";
import { getUserProfile } from "../redux/actions/userActions";
import { getAllUsers, deleteUser } from "../redux/actions/adminActions";
import notifications from "../assets/images/notifications.svg";
import users from "../assets/images/users.svg";
import reported from "../assets/images/reported.svg";
import articleIcon from "../assets/images/articlesIcon.svg";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";
class AdminPage extends Component {
  async componentDidMount() {
    await this.props.getUserProfile(username);
    await this.props.getAllUsers();
  }

  onDelete(id) {
    this.props.deleteUser(id);
    setTimeout(() => this.props.getAllUsers(), 500);
  }

  render() {
    let i = 0;
    return (
      <>
        <Navbar
          profile={this.props.profile}
          token={token}
          signOut={this.signOut}
        />
        <div className="admin-page">
          <Sidebar profile={this.props.profile} />
          <div className="admin-main">
            {this.props.admin.allUsers.length !== 0 ? (
              <>
                <Card
                  cardType="Users"
                  cardNumber={`${this.props.admin.allUsers.usersCount}`}
                  cardIcon={users}
                  cardColor="#D58E8E"
                />
                <Card
                  cardType="Reported"
                  cardNumber={`${this.props.admin.allUsers.reportCount}`}
                  cardIcon={reported}
                  cardColor="#3FAABE"
                />
                <Card
                  cardType="Notifications"
                  cardNumber={`${this.props.admin.allUsers.notifyCount}`}
                  cardIcon={notifications}
                  cardColor="#527667"
                />
                <Card
                  cardType="Articles"
                  cardNumber={`${this.props.admin.allUsers.articleCount}`}
                  cardIcon={articleIcon}
                  cardColor="#959386"
                />
              </>
            ) : null}
            <div className="admin-table">
              <table className="table">
                <thead>
                  <tr>
                    <th>No</th>
                    <th>Names</th>
                    <th>Usernames</th>
                    <th>Email</th>
                    <th>Verified</th>
                    <th>Role</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  {this.props.admin.allUsers.users
                    ? this.props.admin.allUsers.users.map(user => (
                        <tr key={user.id}>
                          <td>{(i += 1)}</td>
                          <td>
                            {user.firstname === null && user.lastname === null
                              ? `${user.firstname} ${user.lastname}`
                              : `No names`}
                          </td>
                          <td>{user.username}</td>
                          <td>{user.email}</td>
                          <td>{`${user.isVerified}`}</td>
                          <td>{user.role}</td>
                          <td>
                            <button
                              onClick={() => this.onDelete(user.id)}
                              style={{ backgroundColor: "white" }}
                            >
                              <i
                                className="fa fa-trash"
                                style={{ color: "red" }}
                              />
                            </button>
                          </td>
                        </tr>
                      ))
                    : null}
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </>
    );
  }
}

AdminPage.propTypes = {
  getUserProfile: PropTypes.func.isRequired,
  profile: PropTypes.object.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  admin: PropTypes.object.isRequired,
  deleteUser: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    profile: state.profile,
    admin: state.admin
  };
};

const mapDispatchToProps = {
  getUserProfile,
  getAllUsers,
  deleteUser
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage);
