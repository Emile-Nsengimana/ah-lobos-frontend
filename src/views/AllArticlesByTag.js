/* eslint-disable max-len */
/* eslint-disable camelcase */
import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import jwt from "jsonwebtoken";
import Navbar from "../components/common/Navbar";
import {
  getUserProfile,
  editUserProfile,
  editUserProfilePicture,
  logOutUser
} from "../redux/actions/userActions";
import { getArticles } from "../redux/actions/articlesAction";
import { getItemDataFromDatabase } from "../helpers/ItemFromEditor/getItemFromEditor";
import ArticlesByTag from "../components/articles/ArticlesByTag";
import Pagination from "../components/Pagination/Pagination";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";

export class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allArticles: [],
      currentPage: 1,
      articlePerPage: 8
    };
  }

  async componentDidMount() {
    await this.props.getUserProfile(username);
    await this.props.getArticles();
  }

  componentWillReceiveProps(newProps) {
    if (newProps.articles.articles) {
      this.setState({ allArticles: newProps.articles.articles });
    }
  }

  signOut = async () => {
    await this.props.logOutUser(token);
  };

  next = () => {
    this.setState({ currentPage: this.state.currentPage + 1 });
  };

  previous = () => {
    this.setState({ currentPage: this.state.currentPage - 1 });
  };

  paginate = pageNumber => {
    this.setState({ currentPage: pageNumber });
  };

  render() {
    const { allArticles, currentPage, articlePerPage } = this.state;
    const indexOfLastArticle = currentPage * articlePerPage;
    const indexOfFirstArticle = indexOfLastArticle - articlePerPage;
    const currentArticles = allArticles.slice(
      indexOfFirstArticle,
      indexOfLastArticle
    );
    const url = window.location.pathname.split("/");
    const tagName = url[url.length - 1];
    if (currentArticles) {
      return (
        <>
          <Navbar
            profile={this.props.profile}
            token={token}
            signOut={this.signOut}
          />
          <div className="tag-article-container">
            <br />
            <div className="tags-container">
              <h4 style={{ color: "grey" }}>TAGGED IN</h4>
              <h1 className="tagName">{decodeURI(tagName)}</h1>
            </div>
            <hr
              style={{
                width: "75%",
                marginLeft: "6%",
                color: "grey"
              }}
            />
            {currentArticles.map(article => {
              if (article.tagList.includes(decodeURI(tagName))) {
                return (
                  <ArticlesByTag
                    key={article.id}
                    article={getItemDataFromDatabase(article)}
                    data={article}
                  />
                );
              }
              return null;
            })}
          </div>
          <Pagination
            articlesPerPage={this.state.articlePerPage}
            totalArticles={this.state.allArticles.length}
            paginate={this.paginate}
            next={this.next}
            previous={this.previous}
            currentPage={this.state.currentPage}
          />
        </>
      );
    }
    return <>Loading...</>;
  }
}
HomePage.propTypes = {
  getUserProfile: propTypes.func.isRequired,
  profile: propTypes.object.isRequired,
  logOutUser: propTypes.func,
  getArticles: propTypes.func.isRequired
};
const mapStateToProps = state => ({
  profile: state.profile,
  articles: state.articles
});
export default connect(
  mapStateToProps,
  {
    getUserProfile,
    editUserProfile,
    editUserProfilePicture,
    getArticles,
    logOutUser
  }
)(HomePage);
