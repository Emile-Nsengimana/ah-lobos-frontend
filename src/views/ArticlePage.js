import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import * as articleActions from "../redux/actions/articleActions";

class ArticlePage extends Component {
  state = {
    article: {
      title: ""
    }
  };

  handleChange = event => {
    const article = { ...this.state.article, title: event.target.value };
    this.setState({ article });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.actions.createArticle(this.state.article);
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h2>Articles</h2>
        <h3>Add Article</h3>
        <input
          type="text"
          onChange={this.handleChange}
          value={this.state.article.title}
        />
        <input type="submit" value="save" />

        {this.props.articles.map(article => (
          <div key={article.title}>{article.title}</div>
        ))}
      </form>
    );
  }
}

ArticlePage.propTypes = {
  actions: PropTypes.object.isRequired,
  articles: PropTypes.array.isRequired
};

function mapStateToProps(state) {
  return {
    articles: state.articles
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(articleActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ArticlePage);
