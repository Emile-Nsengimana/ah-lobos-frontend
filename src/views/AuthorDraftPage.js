/* eslint-disable max-len */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable no-restricted-globals */
import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import jwt from "jsonwebtoken";
import { Link } from "react-router-dom";
import Navbar from "../components/common/Navbar";
import articleImage from "../assets/images/article.png";
import { getUserProfile } from "../redux/actions/userActions";
import {
  getAuthorArticles,
  deleteArticle
} from "../redux/actions/articleActions";
import { getItemDataFromDatabase } from "../helpers/ItemFromEditor/getItemFromEditor";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";

class AuthorArticlesPage extends Component {
  async componentDidMount() {
    await this.props.getUserProfile(username);
    if (this.props.profile.username === "") {
      location.href = "/";
    }
    await this.props.getAuthorArticles();
  }

  onDelete(slug) {
    this.props.deleteArticle(slug);
    setTimeout(() => this.props.getAuthorArticles(), 500);
  }

  render() {
    const checkDraftFound = [];
    if (this.props.myArticles.articles) {
      return (
        <>
          <Navbar profile={this.props.profile} token={token} />
          <div className="articles-container">
            <div className="article-head">
              <h1 style={{ color: "grey" }}>My drafts</h1>
            </div>
            {this.props.myArticles.articles.map(article => {
              const data = getItemDataFromDatabase(article);
              if (article.draft) {
                checkDraftFound.push(article.draft);
                return (
                  <div className="user-articles" key={article.id}>
                    <div className="article-side">
                      <Link
                        to={`/articles/${article.slug}`}
                        className="author-article-title"
                      >
                        <h1 className="article-side-title">{article.title}</h1>
                      </Link>
                      <p className="article-side-desc">
                        {`${article.description.substr(0, 220)}...`}
                      </p>
                      <p className="article-side-footer">
                        - {article.author.username}{" "}
                        <span style={{ float: "right" }}>
                          <Link
                            to={`/article/${article.slug}/edit`}
                            className="edit-article"
                          >
                            <i className="fa fa-edit" />
                          </Link>{" "}
                          <button
                            onClick={() => this.onDelete(article.slug)}
                            style={{ backgroundColor: "white" }}
                          >
                            <i
                              className="fa fa-trash"
                              style={{ color: "red" }}
                            />
                          </button>
                        </span>
                      </p>
                    </div>
                    <div className="article-image">
                      <img
                        src={data.image || articleImage}
                        alt=""
                        style={{
                          height: "200px",
                          width: "80%",
                          float: "right",
                          objectFit: "fill"
                        }}
                      />
                    </div>
                  </div>
                );
              }
              return null;
            })}
            {checkDraftFound.length === 0 ? (
              <div className="not-found">No draft article found...</div>
            ) : null}
          </div>
        </>
      );
    }
    return (
      <>
        <Navbar profile={this.props.profile} token={token} />
        <div className="article-container">
          <h1 style={{ color: "grey" }}>
            You have not created any draft article yet!
          </h1>
          <Link
            to="/article/new"
            style={{ color: "#0D9BC6", textDecoration: "none" }}
          >
            Create new article
          </Link>
        </div>
      </>
    );
  }
}

AuthorArticlesPage.propTypes = {
  profile: PropTypes.object.isRequired,
  myArticles: PropTypes.object.isRequired,
  getAuthorArticles: PropTypes.func.isRequired,
  getUserProfile: PropTypes.func.isRequired,
  deleteArticle: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    profile: state.profile,
    myArticles: state.articles.myArticles,
    deletedArticle: state.articles.deletedArticle
  };
};

const mapDispatchToProps = {
  getUserProfile,
  getAuthorArticles,
  deleteArticle
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AuthorArticlesPage);
