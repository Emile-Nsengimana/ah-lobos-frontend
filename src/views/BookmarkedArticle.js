/* eslint-disable no-restricted-globals */
import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import Navbar from "../components/common/Navbar";
import { getUserProfile } from "../redux/actions/userActions";
import Bookmarks from "../components/articles/Bookmark";
import {
  getBookmarkedArticles,
  unBookmarkArticle
} from "../redux/actions/articleBookmark";

const token = sessionStorage.getItem("token");
const username = sessionStorage.getItem("username");
export class BookmarkedArticle extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleUnbookmark = this.handleUnbookmark.bind(this);
  }

  async componentDidMount() {
    await this.props.getUserProfile(username);
    await this.props.getBookmarkedArticles();
    if (!token) {
      // eslint-disable-next-line no-restricted-globals
      location.href = "/";
    }
  }

  async handleUnbookmark(slug) {
    await this.props.unBookmarkArticle(slug);
    await this.props.getBookmarkedArticles();
  }

  render() {
    return (
      <>
        <Navbar profile={this.props.profile} token={token} />
        <Bookmarks
          bookmarks={this.props.bookmarks}
          unBookmark={this.handleUnbookmark}
        />
      </>
    );
  }
}
BookmarkedArticle.propTypes = {
  getUserProfile: propTypes.func.isRequired,
  profile: propTypes.object.isRequired,
  getBookmarkedArticles: propTypes.any,
  bookmarks: propTypes.object,
  unBookmarkArticle: propTypes.func
};
const mapStateToProps = state => ({
  profile: state.profile,
  bookmarks: state.bookmarks.bookmarks
});
export default connect(
  mapStateToProps,
  { getUserProfile, getBookmarkedArticles, unBookmarkArticle }
)(BookmarkedArticle);
