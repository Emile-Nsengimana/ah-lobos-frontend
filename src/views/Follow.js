import React, { Component } from "react";
import propTypes from "prop-types";
import jwt from "jsonwebtoken";
import { connect } from "react-redux";
import Navbar from "../components/common/Navbar";
import GetFollowers from "../components/Follow/getFollowers";
import {
  getUserProfile,
  editUserProfile,
  editUserProfilePicture,
  logOutUser
} from "../redux/actions/userActions";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";

export class Follow extends Component {
  async componentDidMount() {
    await this.props.getUserProfile(username);
  }

  signOut = async () => {
    await this.props.logOutUser(token);
  };

  render() {
    return (
      <>
        <Navbar />
        <GetFollowers />
      </>
    );
  }
}
Follow.propTypes = {
  getUserProfile: propTypes.func.isRequired,
  logOutUser: propTypes.func
};
const mapStateToProps = state => ({
  profile: state.profile
});

export default connect(
  mapStateToProps,
  {
    getUserProfile,
    editUserProfile,
    editUserProfilePicture,
    logOutUser
  }
)(Follow);
