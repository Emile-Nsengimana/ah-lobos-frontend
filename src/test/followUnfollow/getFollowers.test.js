/* eslint-disable import/no-named-as-default */
/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { shallow } from "enzyme";
import { Follow } from "../../views/Follow";

const props = {
  isFollowing: false,
  getUserProfile: {}
};

describe("Get follower component", () => {
  const component = shallow(<Follow {...props} />);
  it("renders without errors", () => {
    expect(component).toMatchSnapshot();
  });
});
