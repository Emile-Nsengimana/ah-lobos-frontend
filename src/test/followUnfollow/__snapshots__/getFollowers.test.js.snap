// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get follower component renders without errors 1`] = `ShallowWrapper {}`;
